sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/microsoft/opentelemetry/exporter/azuremonitor/_storage.py

```diff
@@ -129,33 +124,27 @@ def gets(self):
         lease_deadline = _fmt(now)
         retention_deadline = _fmt(now - _seconds(self.retention_period))
         timeout_deadline = _fmt(now - _seconds(self.write_timeout))
-        for name in sorted(os.listdir(self.path)):
-            path = os.path.join(self.path, name)
-            if not os.path.isfile(path):
-                continue  # skip if not a file
-            if path.endswith(".tmp"):
-                if name < timeout_deadline:
-                    try:
+        try:
+            for name in sorted(os.listdir(self.path)):
+                path = os.path.join(self.path, name)
+                if not os.path.isfile(path):
+                    continue  # skip if not a file
+                if path.endswith(".tmp"):
+                    if name < timeout_deadline:
                         os.remove(path)  # TODO: log data loss
-                    except Exception:
-                        pass  # keep silent
-            if path.endswith(".lock"):
-                if path[path.rindex("@") + 1 : -5] > lease_deadline:
-                    continue  # under lease
-                new_path = path[: path.rindex("@")]
-                try:
+                if path.endswith(".lock"):
```

Comment:
I believe we want to keep separate logic branches because we might want to log certain outcomes of the maintenance routine in the future.